### PR TITLE
Only attempt to connect to snapshot service once

### PR DESCRIPTION
**Pull request by maxnelso**
If the snapshot service isn't available, this can consume a lot of resources attempting to reconnect.

### CODE REVIEW

File: google_guest_agent/snapshot_listener.go [hunk 1]
```diff
@@ -85,8 +85,8 @@ func listenForSnapshotRequests(address string, requestChan chan<- *sspb.GuestMes
 		logger.Infof("Attempting to connect to snapshot service at %s.", address)
 		conn, err := grpc.Dial(address, grpc.WithInsecure())
 		if err != nil {
-			logger.Errorf("Failed to connect: %v.", err)
-			continue
+			logger.Errorf("Failed to connect to snapshot service: %v.", err)
+			break
```
Comment (hopkiw) [inline google_guest_agent/snapshot_listener.go:89]: altho it works bc the function is entirely one for loop, i think the intention here is maybe better captured by 'return'?

Comment (maxnelso) [reply google_guest_agent/snapshot_listener.go]: Fixed (in both places)!

**Comment by hopkiw** [comment]
ugh
/approve

**Pull request merged**